src/lib/state.ts

```diff
@@ -38,15 +43,33 @@ function getNodeValues(node: ZWaveNode): ValueState[] {
 
 export const dumpValue = (
   node: ZWaveNode,
-  valueId: TranslatedValueID
+  valueArgs:
+    | ZWaveNodeValueAddedArgs
+    | ZWaveNodeValueUpdatedArgs
+    | ZWaveNodeValueRemovedArgs
+    | ZWaveNodeValueNotificationArgs
+    | TranslatedValueID
 ): ValueState => {
-  const valueState = valueId as ValueState;
-  valueState.metadata = node.getValueMetadata(valueId);
-  valueState.value = node.getValue(valueId);
-  // get CC Version for this endpoint, fallback to CC version of the node itself
-  valueState.ccVersion =
-    node.getEndpoint(valueId.endpoint)?.getCCVersion(valueId.commandClass) ||
-    node.getEndpoint(0).getCCVersion(valueId.commandClass);
+  const valueState: ValueState = {
+    commandClassName: valueArgs.commandClassName,
+    commandClass: valueArgs.commandClass,
+    property: valueArgs.property,
+    metadata: node.getValueMetadata(valueArgs),
+    // get CC Version for this endpoint, fallback to CC version of the node itself
+    ccVersion:
+      node
+        .getEndpoint(valueArgs.endpoint)
+        ?.getCCVersion(valueArgs.commandClass) ||
+      node.getEndpoint(0).getCCVersion(valueArgs.commandClass),
+  };
+
+  // make sure that value attribute always holds correct value
+  if ("newValue" in valueArgs) {
+    valueState.value = valueArgs.newValue;
+  } else if ("value" in valueArgs) {
```

Comment:
small typo, should be !"value" in valueArgs. Only if there is value in the valueArgs we retrieve it with node.getValue

Reply:
adjusted